File: test-query.py
#!/usr/bin/env python3
"""
Test script for RAG Microservice query endpoint
"""
import requests
import json

url = "https://ragmicroservice-production.up.railway.app/api/v1/query"

payload = {
    "query": "What is Eden Levi's role?",
    "tenant_id": "default.local",
    "context": {
        "user_id": "admin-user-123",
        "role": "admin"
    }
}

headers = {
    "Content-Type": "application/json"
}

print("Sending request to:", url)
print("Payload:", json.dumps(payload, indent=2))
print("\n" + "="*50 + "\n")

try:
    response = requests.post(url, json=payload, headers=headers, timeout=30)
    print(f"Status Code: {response.status_code}")
    print(f"Response Headers: {dict(response.headers)}")
    print("\nResponse Body:")
    print(json.dumps(response.json(), indent=2))
except requests.exceptions.Timeout:
    print("ERROR: Request timed out after 30 seconds")
except requests.exceptions.RequestException as e:
    print(f"ERROR: {e}")
except json.JSONDecodeError:
    print("Response (not JSON):")
    print(response.text)

File: test-query.js
/**
 * Test script for RAG Microservice query endpoint
 */
import https from 'https';

const url = 'https://ragmicroservice-production.up.railway.app/api/v1/query';

const payload = JSON.stringify({
  query: "What is Eden Levi's role?",
  tenant_id: "default.local",
  context: {
    user_id: "admin-user-123",
    role: "admin"  // This should override any profile role
  }
});

const options = {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': Buffer.byteLength(payload)
  },
  timeout: 30000
};

console.log('Sending request to:', url);
console.log('Payload:', payload);
console.log('\n' + '='.repeat(50) + '\n');

const req = https.request(url, options, (res) => {
  console.log(`Status Code: ${res.statusCode}`);
  console.log(`Response Headers:`, res.headers);
  console.log('\nResponse Body:');
  
  let data = '';
  
  res.on('data', (chunk) => {
    data += chunk;
  });
  
  res.on('end', () => {
    try {
      const json = JSON.parse(data);
      console.log(JSON.stringify(json, null, 2));
    } catch (e) {
      console.log(data);
    }
  });
});

req.on('error', (error) => {
  console.error('ERROR:', error.message);
});

req.on('timeout', () => {
  console.error('ERROR: Request timed out after 30 seconds');
  req.destroy();
});

req.write(payload);
req.end();

